src/test/java/teammates/test/cases/webapi/RestoreFeedbackSessionActionTest.java

```diff
@@ -23,13 +30,69 @@ protected String getRequestMethod() {
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        FeedbackSessionAttributes firstFeedbackSession = typicalBundle.feedbackSessions.get("session1InCourse1");
+
+        String instructorId = instructor1OfCourse1.googleId;
+        String courseId = instructor1OfCourse1.courseId;
+        String feedbackSessionName = firstFeedbackSession.getFeedbackSessionName();
+
+        loginAsInstructor(instructorId);
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, courseId,
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, feedbackSessionName,
+        };
+
+        ______TS("Not found in recycle bin");
+
+        RestoreFeedbackSessionAction notFoundAction = getAction(submissionParams);
+        EntityNotFoundException notFoundException = assertThrows(EntityNotFoundException.class, () -> {
+            getJsonResult(notFoundAction);
+        });
+
+        assertEquals("Feedback session is not in recycle bin", notFoundException.getMessage());
+
+        ______TS("Typical case, restore a deleted feedback session from recycle bin");
+
+        logic.moveFeedbackSessionToRecycleBin(feedbackSessionName, courseId);
+
+        RestoreFeedbackSessionAction action = getAction(submissionParams);
+        JsonResult result = getJsonResult(action);
+        FeedbackSessionData feedbackSessionMessage = (FeedbackSessionData) result.getOutput();
+
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        assertEquals(courseId, feedbackSessionMessage.getCourseId());
+        assertEquals(feedbackSessionName, feedbackSessionMessage.getFeedbackSessionName());
+        assertFalse(logic.getFeedbackSession(feedbackSessionName, courseId).isSessionDeleted());
```

Comment:
It could be a good practice to separate verifying output and verifying model by an empty line, or add comment, but it's fine to me and just a suggestion